builtin/providers/azurerm/resource_arm_network_interface.go

```diff
@@ -0,0 +1,393 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceArmNetworkInterface returns the *schema.Resource
+// associated to network interface resources on ARM.
+func resourceArmNetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmNetworkInterfaceCreate,
+		Read:   resourceArmNetworkInterfaceRead,
+		Update: resourceArmNetworkInterfaceCreate,
+		Delete: resourceArmNetworkInterfaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"vm_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"mac_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"network_security_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"ip_config": &schema.Schema{
```

Comment:
@phinze: concerning your suggestions to move some of these sub-resources to the top-level: in this particular case, and I'd suspect many more, Creating a network interface without defining at least one such IP configuration leads to a 400.

Also, as I'll point out in the test config; most of these sub-resources have `id` fields as well and can/must be used in conjunction with other resource/sub-resources.
As such; we may be able to get away with separating some resources; but most will likely turn out impractical to separate.